sway/commands/bar/colors.c

```diff
@@ -16,38 +18,58 @@ static struct cmd_handler bar_colors_handlers[] = {
 	{ "urgent_workspace", bar_colors_cmd_urgent_workspace },
 };
 
+static bool hex_to_rgba_hex(const char *hex, char rgba[static 10]) {
+	uint32_t color;
+	if (!parse_color(hex, &color)) {
+		return false;
+	}
+	snprintf(rgba, 10, "#%08x", color);
+	return true;
+}
+
 static struct cmd_results *parse_single_color(char **color,
 		const char *cmd_name, int argc, char **argv) {
 	struct cmd_results *error = NULL;
 	if ((error = checkarg(argc, cmd_name, EXPECTED_EQUAL_TO, 1))) {
 		return error;
 	}
-	if (!*color && !(*color = malloc(10))) {
-		return NULL;
+
+	char buffer[10] = {0};
+	if (!hex_to_rgba_hex(argv[0], buffer)) {
+		return cmd_results_new(CMD_INVALID, "Invalid color: %s", argv[0]);
 	}
-	error = add_color(*color, argv[0]);
-	if (error) {
-		return error;
+
+	if (!*color && !(*color = malloc(sizeof(buffer)))) {
```

Comment:
We can just use `strdup` here

Reply:
`hex_to_rgba_hex` could return the dup'ed string btw, this would keep the static buffer + dup logic inside this function.

Reply:
Moved the malloc inside `hex_to_rgba_hex` and removed the static buffers